### DISCUSSION TITLE
Waybar boots ugly upon system start then fixes when the theme is changed

### DISCUSSION THREAD

**Discussion by CFdefense** [prasanthrangan/hyprdots#1892] [Q&A] [answered]
Anyone know why my waybar boots into an ugly config when I boot?
![image](https://github.com/user-attachments/assets/23b16c4e-d4c3-440a-9f1d-581a0be26947)

**Comment by kRHYME7** [comment]
comparison please

**Reply by CFdefense** [reply]
> comparison please

Here is a short video referencing the issue. Upon boot the first waybar is the ugly one I am referencing, and only when switching to certain themes does this issue resolve. It is very strange because switching back to previous themes does not revert back to the ugly config.

[HERE](https://imgur.com/a/8pjMrif) is the video

**Reply by kRHYME7** [reply]
try running `Hyde reload ` and maybe you can catch some errors

**Reply by CFdefense** [reply]
> try running `Hyde reload ` and maybe you can catch some errors

Hyde Reload fixes the ugly waybar I could always try exec on boot as temp fix. What do you think about that?

**Reply by kRHYME7** [reply]
Nah don't . did you try to reboot after running that command?

**Reply by CFdefense** [reply]
> Nah don't . did you try to reboot after running that command?

Just did the reboot and I am still seeing the initial ugly config

**Reply by kRHYME7** [reply]
Hmmm can I see the tail of wbarconfgen.sh script it should be in ~/.local/share/bin

**Reply by CFdefense** [reply]
> Hmmm can I see the tail of wbarconfgen.sh script it should be in ~/.local/share/bin

By tail if you mean the end of the file its
![image](https://github.com/user-attachments/assets/d096f0af-554a-4b68-853d-f22de9754fbf)

**Reply by kRHYME7** [reply]
Looks fine.

Can you please run waybar in the Terminal and show the output. My guess is you have `~/ . config/waybar/config` file which is the priority when you run waybar command alone.

: do it after rebooting

**Comment by CFdefense** [comment]
I ran it and piped the output to here
[output.txt](https://github.com/user-attachments/files/17190647/output.txt)

**Reply by CFdefense** [reply]
Running this waybar always produces the ugly bar

**Reply by kRHYME7** [reply]
Cool Just delete `/home/cfdefence/.config/waybar/config ` for some reasons uou got this configuration. Maybe you installed something before

**Reply by CFdefense** [reply]
That is my linked github repository for my dot files. Why does this cause an issue?

**Reply by CFdefense** [reply]
> That is my linked github repository for my dot files. Why does this cause an issue?

I used stow to link them 
![image](https://github.com/user-attachments/assets/43fde59a-2537-465b-ac35-ab6c85dc72ad)

**Reply by CFdefense** [reply]
Also moving it and running waybar again gives an even uglier waybar 
![image](https://github.com/user-attachments/assets/3fabde3a-d980-4fbc-adf2-fb923272a0d5)

**Reply by kRHYME7** [reply]
It's not finding the config.json? That's strange.

**Reply by kRHYME7** [reply]
try to run wbarconfgen.sh then run waybar see the logs what config it is using

**Reply by kRHYME7** [reply]
try to run wbarconfgen.sh then run waybar see the logs what config it is using

**Reply by CFdefense** [reply]
> try to run wbarconfgen.sh then run waybar see the logs what config it is using

like this? 
![image](https://github.com/user-attachments/assets/20febdd7-4e05-4a6f-a477-68c3c6d748c8)

**Reply by kRHYME7** [reply]
![Screenshot_2024-10-01-09-12-35-017_com.android.chrome-edit.jpg](https://github.com/user-attachments/assets/d36308f6-7a16-405c-8490-4801c1ba5f4f)


waybar is using that config. If you remove that wayabr will use the config.jsonc one.

**Reply by CFdefense** [reply]
> ![Screenshot_2024-10-01-09-12-35-017_com.android.chrome-edit.jpg](https://private-user-images.githubusercontent.com/53417443/372289131-d36308f6-7a16-405c-8490-4801c1ba5f4f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3NDU4NDMsIm5iZiI6MTcyNzc0NTU0MywicGF0aCI6Ii81MzQxNzQ0My8zNzIyODkxMzEtZDM2MzA4ZjYtN2ExNi00MDVjLTg0OTAtNDgwMWMxYmE1ZjRmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAxVDAxMTkwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1YzIxMzA4ODBkNTczMjg1MmQ5ZWQ3Nzc1Yzc3ZTQxODA2ZGZlMjVkZGExNGU4ODMxNzQxY2JkYTkxN2VlY2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Uky_5jxeK4iySxm0lgRJAWq2rSAv8ldmC5TrhGrYH3c)
> 
> waybar is using that config. If you remove that wayabr will use the config.jsonc one.

I’ll try that

**Reply by CFdefense** [reply]
> > ![Screenshot_2024-10-01-09-12-35-017_com.android.chrome-edit.jpg](https://private-user-images.githubusercontent.com/53417443/372289131-d36308f6-7a16-405c-8490-4801c1ba5f4f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3NDU4NDMsIm5iZiI6MTcyNzc0NTU0MywicGF0aCI6Ii81MzQxNzQ0My8zNzIyODkxMzEtZDM2MzA4ZjYtN2ExNi00MDVjLTg0OTAtNDgwMWMxYmE1ZjRmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAxVDAxMTkwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1YzIxMzA4ODBkNTczMjg1MmQ5ZWQ3Nzc1Yzc3ZTQxODA2ZGZlMjVkZGExNGU4ODMxNzQxY2JkYTkxN2VlY2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Uky_5jxeK4iySxm0lgRJAWq2rSAv8ldmC5TrhGrYH3c)
> > waybar is using that config. If you remove that wayabr will use the config.jsonc one.
> 
> I’ll try that

I removed the two config files in that directory and now It is using Using configuration file /etc/xdg/waybar/config.jsonc which is still ugly

**Reply by kRHYME7** [reply]
waybar should find this file 

in my case my username here 
![image](https://github.com/user-attachments/assets/5c2a1ef1-5580-458d-a2be-716b71ccf893)


![image](https://github.com/user-attachments/assets/a244c304-89eb-4308-afbc-82c5dbb35081)


If you can't get it working you can change the exec part of the hyprland.conf into  like this 
` exec-once =  waybar -c ~/.config/waybar/config.jsonc ` I'm just ocnfused why waybar is different in your end

**Reply by kRHYME7** [reply, marked as answer]
> I removed the two config files in that directory and now It is using Using configuration file /etc/xdg/waybar/config.jsonc which is still ugly

Cause you also remove the config.json. Regenerate that with ` wbarconfgen.sh  `

**Reply by CFdefense** [reply]
> > I removed the two config files in that directory and now It is using Using configuration file /etc/xdg/waybar/config.jsonc which is still ugly
> 
> Cause you also remove the config.json. Regenerate that with `wbarconfgen.sh `

Removing config and reloading the config with wbarconfgen made it use the correct .jsonc file. Thanks for all your help, it looks very beautiful now.
